internal/limits.go

```diff
@@ -118,15 +137,35 @@ func (i *Internal) validateJob(job *model.Job) error {
 	// Verify that the user hasn't exceeded their limit for the number of concurrent jobs.
 	jobCount, err := i.countJobsForUser(user)
 	if err != nil {
-		return errors.Wrapf(err, "unable to determine the number of jobs the %s is currently running", user)
+		return http.StatusInternalServerError, errors.Wrapf(err, "unable to determine the number of jobs the %s is currently running", user)
 	}
 	jobLimit, err := i.getJobLimitForUser(user)
 	if err != nil {
-		return errors.Wrapf(err, "unable to determine the concurrent job limit for %s", user)
+		return http.StatusInternalServerError, errors.Wrapf(err, "unable to determine the concurrent job limit for %s", user)
+	}
+	defaultJobLimit, err := i.getDefaultJobLimit()
+	if err != nil {
+		return http.StatusInternalServerError, errors.Wrapf(err, "unable to determine the default concurrent job limit")
 	}
-	if jobCount >= jobLimit {
-		return fmt.Errorf("%s is already running %d or more concurrent jobs", user, jobLimit)
+	var effectiveJobLimit int
+	if jobLimit != nil {
+		effectiveJobLimit = *jobLimit
+	} else {
+		effectiveJobLimit = defaultJobLimit
+	}
+
+	// Return a detailed error if the user has exceeded thier job limit.
+	if jobCount >= effectiveJobLimit {
+		return http.StatusBadRequest, common.ErrorResponse{
+			Message:   fmt.Sprintf("%s is already running %d or more concurrent jobs", user, jobLimit),
+			ErrorCode: "ERR_LIMIT_EXCEEDED",
```

Comment:
Cool, the UI can start checking for this error code 👍 
The only other thing is would it make sense to have an error code like this for when they've exceeded their limit and a separate one for when they haven't yet been approved i.e. their limit is 0 (or some other way to indicate a difference)?  I think the UI needs one message for users with a 0 limit to let them know to fill out the one-time approval form.  And then for users with a limit greater than 0, they can just bug us in Intercom to increase their limit because the assumption is they've already been vetted. 
Or if having 2 different error codes doesn't make sense, the UI can just parse the message string and check the job limit there. Either way is fine by me 👍

Reply:
Having a separate error code seems reasonable to me. I'll add that.

Reply:
Thanks! 🙇